packages/desktop-client/src/components/reports/Reports.js

```diff
@@ -0,0 +1,454 @@
+      let { filters } = await send('make-filters-from-conditions', {
+        conditions,
+      });
+
+      let filte = [...filters];
+      setFilt(filte);
+      setFilterz(conditions);
+    } else {
+      setFilt([]);
+      setFilterz(conditions);
+    }
+  }
+
+  function onChangeDates(start, end) {
+    const numDays = d.differenceInCalendarDays(
+      d.parseISO(end),
+      d.parseISO(start),
+    );
+    const isConcise = numDays > 31 * 3;
+
+    let endDay = end + '-31';
+    if (endDay > monthUtils.currentDay()) {
+      endDay = monthUtils.currentDay();
+    }
+
+    setStart(start + '-01');
+    setEndDay(endDay);
+    setIsConcise(isNetWorth || isIE ? true : isConcise);
+  }
+
+  if (!allMonths || !data) {
+    return null;
+  }
+
+  const handleChange = e => {
+    setSelectList(e.target.value);
+  };
+
+  const handleClick = e => {
+    if (
+      e.target.textContent === 'Totals' ||
+      e.target.id === 'TotalsChoice' ||
+      e.target.parentElement.id === 'TotalsChoice'
+    ) {
+      setIsTotals(true);
+      setSelectList('Expense');
+    } else {
+      setIsTotals(false);
+    }
+
+    e.target.textContent === 'Trends' ||
+    e.target.id === 'TrendsChoice' ||
+    e.target.parentElement.id === 'TrendsChoice'
+      ? setIsTrends(true)
+      : setIsTrends(false);
+  };
+
+  const handleMouseHoverHeader = e => {
+    if (isNetWorth && e.target.id !== 'netWorthHeader') {
+      e.target.style.color = colors.b4;
+      e.target.style.cursor = 'pointer';
+    }
+    if (isCashFlow && e.target.id !== 'cashFlowHeader') {
+      e.target.style.color = colors.b4;
+      e.target.style.cursor = 'pointer';
+    }
+    if (isIE && e.target.id !== 'iEHeader') {
+      e.target.style.color = colors.b4;
+      e.target.style.cursor = 'pointer';
+    }
+  };
+
+  const handleMouseLeaveHeader = e => {
+    if (isNetWorth && e.target.id !== 'netWorthHeader') {
+      e.target.style.color = colors.n7;
+      e.target.style.cursor = 'inherit';
+    }
+    if (isCashFlow && e.target.id !== 'cashFlowHeader') {
+      e.target.style.color = colors.n7;
+      e.target.style.cursor = 'inherit';
+    }
+    if (isIE && e.target.id !== 'iEHeader') {
+      e.target.style.color = colors.n7;
+      e.target.style.cursor = 'inherit';
+    }
+  };
+
+  const onHeaderClick = (e, disable) => {
+    //setIsTrends(false);
+
+    e.target.style.cursor = 'inherit';
+
+    document.getElementById('netWorthHeader').style.color = colors.n7;
+    document.getElementById('cashFlowHeader').style.color = colors.n7;
+    document.getElementById('iEHeader').style.color = colors.n7;
+
+    document.getElementById(e.target.id).style.color = colors.n3;
+
+    setDisableFilter(disable);
+
+    e.target.id === 'netWorthHeader'
+      ? setIsNetWorth(true)
+      : setIsNetWorth(false);
+
+    e.target.id === 'cashFlowHeader'
+      ? setIsCashFlow(true)
+      : setIsCashFlow(false);
+
+    e.target.id === 'iEHeader' ? setIsIE(true) : setIsIE(false);
+  };
```

Comment:
Please remove all of the DOM manipulation here.

Reply:
Done.